File: android/app/src/main/java/itu/mbds/vacataire/models/Professeur.java
package itu.mbds.vacataire.models;

import java.time.LocalDate;
import java.time.LocalTime;
import java.time.format.DateTimeFormatter;
import java.util.List;
import java.util.Locale;
import java.util.stream.Collectors;

public class Professeur {
    public String username;
    public String email;
    public String nom;
    public String prenom;
    public List<String> roles;
    public List<Matiere> matieres;
    public List<Emargement> emargements;
    public double tauxHoraire;

    public Professeur(ProfesseurResponse p) {
        username = p.username;
        email = p.email;
        nom = p.nom;
        prenom = p.prenom;
        roles = p.roles;
        matieres = p.matieres;
        tauxHoraire = p.tauxHoraire;
        emargements = p.emargements.stream().map(emargementString -> {
           return EmargementString.toEmargement(emargementString);
        }).collect(Collectors.toList());
    }
}

File: web/src/app/model/professeur.ts
import {Matiere} from "./matiere";
import {Emargement} from "./emargement";

export class Professeur{
  username: string;
  nom: string;
  prenom: string;
  tauxHoraire: number;
  email: string;
  phoneNumber: string;
  emargements: Emargement[];
  matieres: Matiere[];

  public constructor(init?:Partial<Professeur>) {
    Object.assign(this,init);
  }
}

export interface ProfesseurCreation{
  username: string;
  nom: string;
  prenom: string;
  tauxHoraire?: number;
  email: string;
  phoneNumber: string;
  matieres: string[];
  roles?: string[];
  password: string;
}
